src/ui.tsx

```diff
@@ -571,133 +910,278 @@ export class UserInterface extends Component<Props, State> {
+                          this.resetScaleAndPan();
+                        }
+                        if (minimapToolTip.name === "Minimise Map") {
+                          this.handleDrawerClose();
+                        }
+                      }
+                    )
+                  }
+                >
+                  <Avatar sizes="large">
+                    <SVG
+                      src={`${minimapToolTip.icon}`}
+                      width="55%"
+                      height="auto"
+                      fill={
+                        this.state.buttonClicked === minimapToolTip.name
+                          ? "#02FFAD"
+                          : null
+                      }
+                    />
+                  </Avatar>
+                </IconButton>
+              </HtmlTooltip>
+            ))}
+
+            {/* Background canvas for the minimap */}
+            <BackgroundCanvas
+              scaleAndPan={{ x: 0, y: 0, scale: 1 }}
+              displayedImage={this.state.displayedImage}
+              canvasPositionAndSize={this.state.minimapPositionAndSize}
+              setCanvasPositionAndSize={this.setMinimapPositionAndSize}
+            />
+            <MinimapCanvas
+              displayedImage={this.state.displayedImage}
+              scaleAndPan={this.state.scaleAndPan}
+              setScaleAndPan={this.setScaleAndPan}
+              canvasPositionAndSize={this.state.viewportPositionAndSize}
+              minimapPositionAndSize={this.state.minimapPositionAndSize}
+              setMinimapPositionAndSize={this.setMinimapPositionAndSize}
+            />
+          </Card>
+        </Slide>
+
+        {this.state.toggleMinimap === false ? (
+          <Slide
+            in={!this.state.toggleMinimap}
+            direction="up"
+            timeout={{ enter: 1000 }}
+          >
+            <Card
+              style={{
+                width: "61px",
+                height: "53px",
+                left: "540px",
+                position: "relative",
+                borderRadius: "10px 0 0 0",
+                textAlign: "center",
+                padding: "7px 0",
+              }}
+            >
+              <HtmlTooltip
+                key="Maximise Map"
+                title={
+                  <Box
+                    display="flex"
+                    alignItems="center"
+                    justifyItems="space-between"
+                  >
+                    <Box mr={3}>
+                      <Typography color="inherit">Maximise Map</Typography>
+                    </Box>
+                    <Avatar
+                      style={{
+                        backgroundColor: "#02FFAD",
+                        color: "#2B2F3A",
+                      }}
+                    >
+                      Alt
+                    </Avatar>
+                    <Avatar
+                      style={{
+                        backgroundColor: "#02FFAD",
```

Comment:
All of these hardcoded colours should be replaced with variables (from the theme?)